### PR TITLE
Support type erasure on task_t

### PR DESCRIPTION
**Pull request by apolyakov**

### CODE REVIEW

**Review by mkornaukhov03 — COMMENTED** [review]
OK, but there are some questions

File: runtime-light/stdlib/fork/fork-api.cpp [hunk 1]
```diff
@@ -15,10 +14,10 @@ task_t<void> f$sched_yield() noexcept {
   co_await wait_for_reschedule_t{};
 }
 
-task_t<void> f$sched_yield_sleep(int64_t duration_ns) noexcept {
-  if (duration_ns < 0) {
-    php_warning("can't sleep for negative duration %" PRId64, duration_ns);
+task_t<void> f$sched_yield_sleep(double duration) noexcept {
+  if (duration <= 0) {
```
[the remark this answers is not in the document]
Comment (apolyakov) [reply runtime-light/stdlib/fork/fork-api.cpp:18]: Exactly

File: runtime-light/stdlib/fork/fork-api.cpp [hunk 2]
```diff
@@ -15,10 +14,10 @@ task_t<void> f$sched_yield() noexcept {
...
+    php_warning("can't sleep for negative or zero duration %.9f", duration);
     co_return;
   }
-  co_await wait_for_timer_t{std::chrono::nanoseconds{static_cast<uint64_t>(duration_ns)}};
+  co_await wait_for_timer_t{std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::duration<double>{duration})};
```
[the remark this answers is not in the document]
Comment (apolyakov) [reply runtime-light/stdlib/fork/fork-api.cpp:22]: I prefer not to use any usings that introduce lots of names into the scope

[the remark this answers is not in the document]
Comment (mkornaukhov03) [reply runtime-light/stdlib/fork/fork-api.cpp:22]: > I prefer not to use any usings that introduce lots of names into the scope

I mean
```c++
using std::chrono::nanoseconds;
using std::chrono::duration_cast;
// and so on
```

**Review by mkornaukhov03 — DISMISSED** [review]
LGTM

**Review by mkornaukhov03 — APPROVED** [review]

**Pull request merged**